lite2/client.go

```diff
@@ -830,19 +851,36 @@ func (c *Client) backwards(
 		}
 
 		if err := interimHeader.ValidateBasic(c.chainID); err != nil {
```

Comment:
the following 3 if conditions need to be extracted into a function. then we won't have to repeat replacePrimaryProvider 3 times

Reply:
I'm going to put it in `verifier.go` unless you disagree